### PR TITLE
Always use AzCopy to get total resources to upload

### PR DESCRIPTION
**Pull request by wwlorey**
Fixes https://github.com/microsoft/vscode-azurestorage/issues/873

### CODE REVIEW

File: src/commands/uploadFolder.ts [hunk 1]
```diff
@@ -48,11 +48,11 @@ export async function uploadFolder(
     try {
         if (notificationProgress && cancellationToken) {
             // AzCopy recognizes folders as a resource when uploading to file shares. So only set `countFoldersAsResources=true` in that case
-            await uploadLocalFolder(actionContext, treeItem, sourcePath, destPath, notificationProgress, cancellationToken, destPath, treeItem instanceof FileShareTreeItem);
+            await uploadLocalFolder(actionContext, treeItem, sourcePath, destPath, notificationProgress, cancellationToken, destPath);
```
Comment (nturinski) [inline src/commands/uploadFolder.ts:51]: Can't you get of the `countFoldersAsResources` parameter now?

**Review by nturinski — COMMENTED** [review]
I'm a little confused as to how this is working.  Isn't this the code that was reporting the progress?  https://github.com/microsoft/vscode-azurestorage/blob/main/src/commands/azCopy/azCopyTransfer.ts#L105-L109  It looks like that this was always using the `TranserStatus.status` preferentially and only relying on the passed in total work if it couldn't find it in the status.

**Comment by wwlorey** [comment]
> Isn't this the code that was reporting the progress? https://github.com/microsoft/vscode-azurestorage/blob/main/src/commands/azCopy/azCopyTransfer.ts#L105-L109

This has been confusing me for a bit lol. So that check was originally to make sure that no progress was shown until we were sure that `totalWork` existed. But you're right that it clobbers whatever `totalWork` may have been passed when creating the `TransferProgress`. So I can totally remove `totalWork` as a class member

**Comment by wwlorey** [comment]
Let me know if you want me to break this up into multiple PRs - that latest commit introduced a lot of changes

**Review by ejizba — APPROVED** [review]

**Pull request merged**